### DISCUSSION TITLE
R22.7.16 Release

### DISCUSSION THREAD

**Discussion by coolsnowwolf** [coolsnowwolf/lede#9773] [Bugs / 问题反馈]
新增加 R5S EMMC 版本支持：

初始 IP 192.168.100.1
增加 2.5G 网口 Zerocopy 支持，大幅降低 CPU 资源跑满带宽

<hr /><em>This discussion was created from the release <a href='https://github.com/coolsnowwolf/lede/releases/tag/'>R22.7.16 Release</a>.</em>

**Comment by iskoldt-X** [comment]
大佬，没有Docker 真的很难受😭

**Reply by fishg** [reply]
用sysup 从web升级安装就有了。。

**Comment by coolsnowwolf** [comment]
哪个版本的 docker ？

**Reply by iskoldt-X** [reply]
电犀牛R68s 的7.16 固件没有带Docker，不知道自己在线安装会不会有性能损失，所以没有贸然行动

**Reply by fishg** [reply]
@iskoldt-X 我想在线装，不知道哪里可找到匹配的ipk，你知道吗？

**Comment by ifdog** [comment]
大佬，自己编译时没有r5s的subtarget呀，应该怎样编译出R5S的固件

**Reply by redchenjs** [reply]
同问 +1

**Reply by ouhaiorg** [reply]
同问 +2

**Reply by fishg** [reply]
同问+3，不知道怎么自己编译. 想用lxc容器跑，可惜缺了kmod-veth。看来还是要学会编译r5s固件

**Comment by zzx** [comment]
r5s固件，掉盘严重。两块usb移动硬盘，路由重启后经常改变挂载点，在sda1和sdb1中不断变动。

**Reply by coolsnowwolf** [reply]
移动硬盘的挂载前需要HDD盘片先初始化，哪个硬盘先初始化成功就哪个是 A

如果有多盘的需求最好手动指定UUID挂载点而不是自动

**Reply by zzx** [reply]
谢谢，我试试。

**Comment by fance068** [comment]
68s 挂一块 移动硬盘 重启就找不到了，不重启倒是很好用 哈哈

**Reply by coolsnowwolf** [reply]
R22.7.19 解决了，直接升级即可

**Reply by fance068** [reply]
感谢大雕

**Comment by npcXxX** [comment]
缺少kmod-fuse 还有 luci-lib-base rpcd-mod-luci 
还有网易云的插件之前就失效了没有更新的话可以去掉了

**Reply by coolsnowwolf** [reply]
luci-lib-base rpcd-mod-luci 不存在于lua 版本的 luci

**Reply by npcXxX** [reply]
> luci-lib-base rpcd-mod-luci 不存在于lua 版本的 luci

那有kmod-fuse吧，rclone没有luci也还是可以用命令行的

**Comment by Suyuan406** [comment]
大雕，RNDIS没有加上喔，能否在下个版本把这个加上？感谢！
![image](https://user-images.githubusercontent.com/73843390/179935960-a5d39e8f-98a3-47e6-8392-d780b2f7539d.png)

**Comment by hienao** [comment]


**Reply by p-w-p** [reply]
https://op.supes.top/packages/aarch64_generic/libmbedtls12_2.28.0-9_aarch64_generic.ipk

**Comment by scyslz** [comment]
![截图_20220723140153](https://user-images.githubusercontent.com/39955047/180592694-55502814-9e93-4ab9-95b6-41f4737dd351.png)
![截图_20220723140244](https://user-images.githubusercontent.com/39955047/180592715-05e0ef63-c132-4845-b742-533e346716a6.png)
openclash 更新不了啊，空间不足啊

**Reply by f0banzai** [reply]
因为下载地址解析问题。手动添加内核。

**Comment by iskoldt-X** [comment]
谢谢大佬的固件，群里7.22 固件里面预装有bonding 的kmod 等插件。我把R68s 与TP-LINK XDR3010 （家境贫寒，买不起带2.5G 口的硬路由）链路聚合连接后，iPhone 使用WebDav 从电牛子下载本地分享的电影时的速度真的变快了，而且更稳定了！

美中不足的是7.22 没有Docker，幸亏我有两个电牛子，第一个直接用Docker 搞的WebDav 文件分享，用2.5G 网口与第二个刷了7.22 的连接，第二个电牛子再链路聚合与TP-LINK 连接。即使是多了一层电牛子，速度也变快了（第一个电牛子无法搞链路聚合）。没有测试samba，因为在iPhone 上，samba 真的比WebDav 慢很多。

诚恳地请求大佬在以后的Docker 版固件里都加入bonding 的一系列插件！🫡

壳子已买

**Comment by pengfanzhi** [comment]
大佬 安装airplay2 没有kmod-sound-core 插hdmi无声音输出

**Reply by coolsnowwolf** [reply]
hdmi audio 被我禁止了，这需求不明显啊

**Comment by Plutonium141** [comment]
大雕 小马m68s的固件还会更新吗

**Reply by coolsnowwolf** [reply]
会的，等出差回来

**Reply by ouhaiorg** [reply]
m68s的下个固件能把kmod-gre编译进去吗，没有这个不能装pptpd

**Reply by fishg** [reply]
gre 同样强烈需求

**Comment by MinamiGua** [comment]
想问下usb接口能不能插蓝牙适配器呀~

**Comment by toraidl** [comment]
大神您好，
电犀牛R68S从7.19升级到7.22后，发现Lan口位置不对， 原先的eth2口，现在变为eth3口，结果就是家里面的eth2连接的小米路由有线mesh显示离线。降级为7.19就好了
![image](https://user-images.githubusercontent.com/3390079/180714943-09ba9667-3763-4b3f-9923-39bcd641de80.png)

**Comment by ostrichii** [comment]
@coolsnowwolf 请教大雕，emmc_update解出来的img文件，能否直接通过原来的friendlywrt 用dd写入emmc分区？

**Reply by pskoma** [reply]
可以

**Comment by fishg** [comment]
希望能带无线驱动：RTL8811CU/RTL8821CU

**Comment by MasterW1900** [comment]
大佬好，请问有支持H68K内置AP6256无线网卡的计划吗？

**Comment by npcXxX** [comment]
8.8docker版本的BusyBox不知道是不是有问题，一直报错_ps: unrecognized option: C_
722版本没有这个问题的

![image](https://user-images.githubusercontent.com/43200321/184075217-b85fbbeb-efa5-414f-9f97-1709812d9382.png)

**Comment by pengfanzhi** [comment]
-R能跑满 正向几百 why?

Server listening on 5201 (test #1)
-----------------------------------------------------------
Accepted connection from 192.168.100.109, port 49501
[  5] local 192.168.100.1 port 5201 connected to 192.168.100.109 port 49502
[ ID] Interval           Transfer     Bitrate         Retr  Cwnd
[  5]   0.00-1.00   sec   278 MBytes  2.34 Gbits/sec    0    588 KBytes       
[  5]   1.00-2.00   sec   281 MBytes  2.36 Gbits/sec    0    594 KBytes       
[  5]   2.00-3.00   sec   280 MBytes  2.35 Gbits/sec    0    594 KBytes       
[  5]   3.00-4.00   sec   281 MBytes  2.36 Gbits/sec    0    624 KBytes       
[  5]   4.00-5.00   sec   280 MBytes  2.35 Gbits/sec    0    656 KBytes       
[  5]   5.00-6.00   sec   280 MBytes  2.35 Gbits/sec    0    687 KBytes       
[  5]   6.00-7.00   sec   281 MBytes  2.36 Gbits/sec    0    687 KBytes       
[  5]   7.00-8.00   sec   280 MBytes  2.35 Gbits/sec    0    720 KBytes       
[  5]   8.00-9.00   sec   280 MBytes  2.35 Gbits/sec    0    720 KBytes       
[  5]   9.00-10.00  sec   281 MBytes  2.36 Gbits/sec    0    720 KBytes       
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-10.00  sec  2.74 GBytes  2.35 Gbits/sec    0             sender
-----------------------------------------------------------
Server listening on 5201 (test #2)
-----------------------------------------------------------
Accepted connection from 192.168.100.109, port 49503
[  5] local 192.168.100.1 port 5201 connected to 192.168.100.109 port 49504
[ ID] Interval           Transfer     Bitrate
[  5]   0.00-1.00   sec  53.7 MBytes   450 Mbits/sec                  
[  5]   1.00-2.00   sec  54.4 MBytes   456 Mbits/sec                  
[  5]   2.00-3.00   sec  53.9 MBytes   453 Mbits/sec                  
[  5]   3.00-4.00   sec  54.1 MBytes   454 Mbits/sec                  
[  5]   4.00-5.00   sec  54.1 MBytes   454 Mbits/sec                  
[  5]   5.00-6.00   sec  54.0 MBytes   453 Mbits/sec                  
[  5]   6.00-7.00   sec  54.0 MBytes   453 Mbits/sec                  
[  5]   7.00-8.00   sec  53.8 MBytes   452 Mbits/sec                  
[  5]   8.00-9.00   sec  53.8 MBytes   451 Mbits/sec                  
[  5]   9.00-10.00  sec  54.2 MBytes   455 Mbits/sec                  
[  5]  10.00-10.03  sec  1.53 MBytes   453 Mbits/sec                  
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bitrate
[  5]   0.00-10.03  sec   542 MBytes   453 Mbits/sec                  receiver
-----------------------------------------------------------
Server listening on 5201 (test #3)

**Comment by ouhaiorg** [comment]
![IMG_20220902_082139](https://user-images.githubusercontent.com/29619015/188076545-daffbca0-51ff-4c52-8030-b11c354aadd2.jpg)
新编译的m68s  sysupgrade固件无法在原来的R22.8.22-xiaoma-m68s-emmc固件上线刷，请问一下如何打包成emmc-update，用rkdeveloptool可以安装的

**Reply by hzjnet** [reply]
大神，能不能修订下ipq401x啊：
全部编译时提示：
A build_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/linux-ipq40xx/qcom_ap-dk04.1-c1-fit-uImage.itb': No such file or directory
B Makefile:823: recipe for target '/xxxx/bin/targets/ipq40xx/generic/-fit-uImage.itb' failed
应该是同一问题，手动后也能正常继续

今天接着尝试记得A是4040的IMAGES += eva.bin问题，改行换成：IMAGES = sysupgrade.bin后编译正常
B的问题应该找-fit-uImage.itb目录建立过程，需要设置断点，还没找到原因

**Comment by npcXxX** [comment]
想问一下以后都在群里发固件或者云盘吗，这里还发更新包吗

**Comment by xaqingfeng** [comment]
大佬，最新一版（ R22.10.10）passwall 软件包可以搜出，但安装不上， 显示缺少依赖

**Comment by menphy1989** [comment]
R68S 11.13版本web和后台均升级不了
root@OpenWrt:/tmp# sysupgrade -F openwrt-rockchip-R22.11.13-fastrhino_r68s-squashfs-sysupgrade.img.bin 
Sat Aug  5 17:04:43 CST 2017 upgrade: Device fastrhino,r68s not supported by this image
Sat Aug  5 17:04:43 CST 2017 upgrade: Supported devices: rockchip,rk3568-lunzn-r68s
Sat Aug  5 17:04:43 CST 2017 upgrade: Reading partition table from bootdisk...
Sat Aug  5 17:04:43 CST 2017 upgrade: Reading partition table from image...
Sat Aug  5 17:04:43 CST 2017 upgrade: Invalid partition table on /tmp/image.bs
Failed to parse message data
sh: out of range
Sat Aug  5 17:04:44 CST 2017 upgrade: Saving config files...
Sat Aug  5 17:04:48 CST 2017 upgrade: Commencing upgrade. Closing all shell sessions.
{
	"error": {
		"message": "Firmware image couldn't be validated: no JSON input"
	}
}
Command failed: Unknown error
![image](https://user-images.githubusercontent.com/12580316/201573620-0ffa736d-19bc-440d-9f2e-74ba87820016.png)

**Comment by f0banzai** [comment]
R5S R22.7.19
PC和NAS均使用2.5G网卡链接至R5S的两个2.5G接口。
使用iperf3分别用PC和NAS，各自与R5S之间测速，均能2Gb+。
而若是PC和NAS之间，则跌到450Mb/s
关闭防火墙(/etc/init.d/firewall stop)后也不过1Gb/s
这个是因为启动了什么功能吗？

**Comment by copperfieldd** [comment]
![JW QLRYY{Y_G}K3JHUR%30](https://user-images.githubusercontent.com/25824246/209760155-e774d7f2-af2b-4476-ad6f-afe0ac525d3e.png)
![image](https://user-images.githubusercontent.com/25824246/209761614-bc929486-b825-42c6-9b0a-9ec2c602efb9.png)

大佬好，NAS  ddns需要用到ipv6公网，目前开启路由ipv6域名解析后pc和手机上外网速度受严重影响了，一个域名同时具备ipv4和ipv6默认会先解析ipv6，请问可以在路由上配置ipv6和ipv4的解析优先级吗？（需要优先解析ipv4，ipv4不能用再解析ipv6）linux下可以修改/etc/gai.conf文件实现，opwrt里面未找到相应配置
label  ::1/128       0
label  ::/0          1
label  2002::/16     2
label ::/96          3
label ::ffff:0:0/96  4
precedence  ::1/128       50
precedence  ::/0          40
precedence  2002::/16     30
precedence ::/96          20
precedence ::ffff:0:0/96  10

**Comment by Jimmisama** [comment]
大佬能顺便H66K吗？我是乞丐1G版本 无WiFi H66K，配置大概就是r66s低配加h68k低配的结合体

**Comment by ZepEthan** [comment]
请问大佬，要怎么编译才能得到EMMC的固件呢？以前只有ext4和update版本的。

**Reply by bdwg2686600** [reply]
同问，请问要怎么编译才能得到  emmc_update 为线刷文件呢？

**Comment by ZachQin** [comment]
能提供一份附件里的固件对应的.config文件吗？

**Comment by 8MiYile** [comment]
大dio哥，有ipk源吗？

**Comment by murphytalk** [comment]
请问现在买到的是所谓2023年版，这个固件也可以直接刷吗
